packages/Webkul/Product/src/Type/AbstractType.php

```diff
@@ -535,7 +535,7 @@ public function getMinimalPrice($qty = null)
      */
     public function getMaximamPrice()
     {
-        return $this->getMinimalPrice();
+        return $this->product->price;
```

Comment:
This can't be product price. This method represents many products type like configurable, grouped. So minimum price is needed.

Reply:
Hello @devansh-webkul can you tell me what am I missing:
1. Go to Webkul/Product/src/Type you have few files for all types of products
2. They all extend the first file AbstractType which has faulty definition for getMaximamPrice that will always return special_price
3. But then you go to Configurable type and see that getMaximamPrice is redefined for this type and does something else:
https://github.com/bagisto/bagisto/blob/1ece839bba6e597f4e380b952cb4188068d59901/packages/Webkul/Product/src/Type/Configurable.php#L432

Also you go to Bundle and see the same thing:
https://github.com/bagisto/bagisto/blob/1ece839bba6e597f4e380b952cb4188068d59901/packages/Webkul/Product/src/Type/Bundle.php#L276


I've tested the solution and it works great, because in the end configurable product return a collection of simple products, so it's correct and doesn't mess up with Configurable and Bundle products.
The only concern is that Grouped is the only type that hasn't defined the getMaximamPrice method, but I don't know what it should be.

So are you suggesting I should return $this->product->price only for Simple product type?

Reply:
In sorting at the home page this will create a problem. Just verify with the grouped products also. Because you have given actual product price. Verify also by adding selling price also.

Reply:
I've made changes and defined the getMaximamPrice only for Simple, Virtual and Downloadable types, grouped should be fine, also sorting uses min price right?

Reply:
Ok, I am assigning this for testing. I Will let you know.

Reply:
Thanks! :)

Also I've changed the category_id key to categories for the product endpoint, this should be included in the docs, or I can change it back to category_id if it will introduce any braking changes?

Reply:
Just check with the web also you have done changes in the repository and this repository also calling the web controller also.

Reply:
in API response, 
if a product has a special price then price should return the normal price but in case of configurable product it was returning the same special price due to getMinimalPrice() method.

Resolved this by .. 
file : \packages\Webkul\API\Http\Resources\Catalog\Product.php

`'price'                  => $product->price ? $product->price : $product->getTypeInstance()->getMinimalPrice(),`

and it's working in all cases